backend-shared/config/db/util/application_info_cache.go

```diff
@@ -0,0 +1,411 @@
+	}
+
+	req.responseChannel <- applicationInfoCacheResponse{
+		err: err,
+	}
+
+}
+
+func processDeleteAppStateMessage(dbQueries db.DatabaseQueries, req applicationInfoCacheRequest, cacheApp map[string]applicationCacheEntry, cacheAppState map[string]applicationStateCacheEntry) {
+
+	if db.IsEmpty(req.primaryKey) {
+		log.Error(fmt.Errorf("PrimaryKey should not be nil"), "SEVERE: "+req.primaryKey+" not found")
+	}
+
+	// Remove from cache
+	delete(cacheAppState, req.primaryKey)
+	delete(cacheApp, req.primaryKey)
+
+	// Remove from DB
+	rowsAffected, err := dbQueries.DeleteApplicationStateById(req.ctx, req.primaryKey)
+
+	req.responseChannel <- applicationInfoCacheResponse{
+		rowsAffectedForDelete: rowsAffected,
+		err:                   err,
+	}
+
+}
+
+func processGetAppStateMessage(dbQueries db.DatabaseQueries, req applicationInfoCacheRequest, cacheApp map[string]applicationCacheEntry, cacheAppState map[string]applicationStateCacheEntry) {
+
+	appState := db.ApplicationState{
+		Applicationstate_application_id: req.primaryKey,
+	}
+
+	var err error
+	var valueFromCache bool
+
+	if db.IsEmpty(req.primaryKey) {
+		log.Error("PrimaryKey should not be nil: " + req.primaryKey + " not found")
+	} else {
+		delete(cacheAppState, req.createOrUpdateAppStateObject.Applicationstate_application_id)
+		delete(cacheApp, req.createOrUpdateAppStateObject.Applicationstate_application_id)
+	}
+
+	res, exists := cacheAppState[appState.Applicationstate_application_id]
+	if !exists {
+		// If it's not in the cache, then get it from the database
+		valueFromCache = false
+		err = dbQueries.GetApplicationStateById(req.ctx, &appState)
+		if err != nil {
+			appState = db.ApplicationState{}
+		}
+
+		// Update the cache if we get a result from the database
+		// Update valueFromCache to false, since data is retrieved from db
+		if err == nil && appState.Applicationstate_application_id != "" {
+			if entryInCache, exists := cacheAppState[appState.Applicationstate_application_id]; !exists {
+				entryInCache.appState = appState
+				cacheAppState[appState.Applicationstate_application_id] = entryInCache
+			}
+		}
+
+	} else {
+		// If it is in the cache, return it from the cache
+		valueFromCache = true
+		appState = res.appState
+	}
+
+	req.responseChannel <- applicationInfoCacheResponse{
+		applicationState: appState,
+		valueFromCache:   valueFromCache,
+		err:              err,
+	}
+
+}
+
+func processGetAppMessage(dbQueries db.DatabaseQueries, req applicationInfoCacheRequest, cacheApp map[string]applicationCacheEntry, cacheAppState map[string]applicationStateCacheEntry) {
+
+	app := db.Application{
+		Application_id: req.primaryKey,
+	}
+
+	var err error
+	var valueFromCache bool
+
+	if db.IsEmpty(req.primaryKey) {
+		log.Error("PrimaryKey should not be nil: " + req.primaryKey + " not found")
+	} else {
+		delete(cacheApp, req.createOrUpdateAppStateObject.Applicationstate_application_id)
```

Comment:
Just like with `processGetAppStateMessage`, I think these deletes can be moved below.